src/tools/pkcs15-init.c

```diff
@@ -2987,9 +2987,13 @@ read_options_file(const char *filename)
 			if (o->has_arg != no_argument) {
 				optarg = strtok(NULL, "");
 				if (optarg) {
+					char *start = optarg;
 					while (isspace((int) *optarg))
 						optarg++;
-					optarg = strdup(optarg);
+					if (optarg != start) {
+						optarg = strdup(optarg);
+						free(start);
+					}
```

Comment:
I don't think this is a correct fix. The issue here is that we have (global?) variable `optarg`, which is set here and possibly handled in `handle_option()` -- assigned to different variable, but in following cycle we are overwriting `strdup`-ed variable with a new one.

Your proposed solution will allow assigning pointers to the static buffer to these options, which will get overwritten in the next cycle.

Reply:
I've removed this commit; I didn't find an easy solution to this.

I think the easiest way to handle this issue is to deprecate `--options-file`, which isn't used in any other tool. It was invented to not expose PINs via `ps`, which we can use environment variables for nowadays...